src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1408,6 +1399,22 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
         addColumnsDescriptionToCreateQueryIfNecessary(query_ptr->as<ASTCreateQuery &>(), res);
     }
 
+    std::unordered_set<std::string_view> all_columns;
+    all_columns.reserve(properties.columns.size());
+
+    for (const auto & column : properties.columns)
+        all_columns.emplace(column.name);
+
+    auto reserved_columns = res->getVirtuals().getNames();
```

Comment:
Can I create a `Distributed` table with `_part UInt8` column, for example? What will happen on an attempt to read from a `MergeTree` table through that distributed table? Please add some tests. Also it makes sense to check a similar case with `Merge` engine

Reply:
StorageDistributed::getVirtuals() has a bunch of strange columns, so for now I disabled checks for it.
https://github.com/ClickHouse/ClickHouse/blob/35e9e45249061afbef618f511a191b150864d291/src/Storages/StorageDistributed.cpp#L290-L305

StorageDistributed seems to check whether its virtual columns are virtual, so I also disabled checks for it.
https://github.com/ClickHouse/ClickHouse/blob/35e9e45249061afbef618f511a191b150864d291/src/Storages/StorageMerge.cpp#L319-L327